### PR TITLE
JIT: add more "simple" complexity helpers

### PR DESCRIPTION
**Pull request by AndyAyersMS**
In early phases we haven't set tree costs or size estimates, so we fall back to counting statements or nodes.

Capture some of this in some helper methods, both pure counting and "exceeds" variants (which are potentially cheaper when comparing against a budget).

Update the loop cloning size check to use this.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@amanasifkhalid PTAL
cc @dontnet/jit-contrib

Should be zero diff. I will add more uses of this subsequently.

File: src/coreclr/jit/block.cpp [hunk 1]
```diff
@@ -1942,3 +1942,134 @@ StackEntry* BasicBlock::bbStackOnEntry() const
...
+//   exceeds some limit
+//
+// Arguments:
+//    limit  - limit on the number of statements
+//    count  - [out, optional] actual number of statements (if less than or equal to limit)
+//
+// Returns:
+//   true if the number of statements is greater than limit
+//
+bool BasicBlock::StatementCountExceeds(unsigned limit, unsigned* count)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/block.cpp:1979]: nit:
```suggestion
bool BasicBlock::StatementCountExceeds(unsigned limit, unsigned* count /* = nullptr */)
```
etc.

File: src/coreclr/jit/block.cpp [hunk 2]
```diff
@@ -1942,3 +1942,134 @@ StackEntry* BasicBlock::bbStackOnEntry() const
...
+    if (count != nullptr)
+    {
+        *count = localCount;
+    }
+
+    return overLimit;
+}
+
+//------------------------------------------------------------------------
+// ComplexityExceeds: check if the number of nodes in the trees in athe block
```
Comment (amanasifkhalid) [inline src/coreclr/jit/block.cpp:2002]:
```suggestion
// ComplexityExceeds: check if the number of nodes in the trees in the block
```

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

File: src/coreclr/jit/block.cpp [hunk 3]
```diff
@@ -1942,3 +1942,134 @@ StackEntry* BasicBlock::bbStackOnEntry() const
...
+//
+// Arguments:
+//    comp   - compiler instance
+//    limit  - limit on the number of nodes
+//    count  - [out, optional] actual number of nodes (if less than or equal to limit)
+//
+// Returns:
+//   true if the number of nodes is greater than limit
+//
+bool BasicBlock::ComplexityExceeds(Compiler* comp, unsigned limit, unsigned* count)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/block.cpp:2013]:
```suggestion
bool BasicBlock::ComplexityExceeds(Compiler* comp, unsigned limit, unsigned* count /* = nullptr */)
```

File: src/coreclr/jit/block.cpp [hunk 4]
```diff
@@ -1942,3 +1942,134 @@ StackEntry* BasicBlock::bbStackOnEntry() const
...
+//
+// Arguments:
+//    comp   - compiler instance
+//    limit  - limit on the number of nodes
+//    count  - [out, optional] actual number of nodes (if less than or equal to limit)
+//
+// Returns:
+//   true if the number of nodes is greater than limit
+//
+bool BasicBlockRangeList::ComplexityExceeds(Compiler* comp, unsigned limit, unsigned* count)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/block.cpp:2051]:
```suggestion
bool BasicBlockRangeList::ComplexityExceeds(Compiler* comp, unsigned limit, unsigned* count /* = nullptr */)
```

**Comment by AndyAyersMS** [comment]
Hmm, there are a few diffs. I didn't see these locally. Let me re-check.

**Comment by AndyAyersMS** [comment]
There was a loop of size exactly 400, which we now clone.

**Pull request merged**